### PR TITLE
feat: add action to build container image

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Extracts the action to build a DCR container into its own PR. Details of why we're doing this in #6951

### CODE REVIEW

**Review by arelra — COMMENTED** [review]
This is going to be super useful for @guardian/commercial-dev as we currently build DCR as part of our e2e tests which takes up the majority of the time for our CI. This will hopefully give us a big speed boost. Thank you 🙏

File: dotcom-rendering/Containerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:16.19.0-alpine AS dependencies
```
Comment (OllysCoding) [inline dotcom-rendering/Containerfile:1]: Can we add a comment highlighting that this is used for CI/CD only, and not deployed as a container to production?

File: .github/workflows/dcr-container.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        working-directory: dotcom-rendering
+
+      - name: Build Container Image
+        id: build_image
+        uses: redhat-actions/buildah-build@v2.11
+        with:
+          image: dotcom-rendering
+          tags: ${{ github.sha }} ${{ env.GITHUB_REF_SLUG }}
+          context: ./
+          containerfiles: ./dotcom-rendering/Containerfile
```
Comment (OllysCoding) [inline .github/workflows/dcr-container.yml:42]: Wonder if we could even rename this to `Containerfile.ci` or something that makes it clear it's not for production?

File: .github/workflows/dcr-container.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Generate production build
+        run: make build
+        working-directory: dotcom-rendering
+
+      - name: Build Container Image
+        id: build_image
+        uses: redhat-actions/buildah-build@v2.11
+        with:
+          image: dotcom-rendering
+          tags: ${{ github.sha }} ${{ env.GITHUB_REF_SLUG }}
```
[the remark this answers is not in the document]
Comment (AshCorr) [reply .github/workflows/dcr-container.yml:40]: Ah yes good point! Forgot that was the case. I think we'll be wanting to use the HEAD commit.

[the remark this answers is not in the document]
Comment (AshCorr) [reply .github/workflows/dcr-container.yml:40]: (ah I'm not actually using the pull_request trigger just yet)

[the remark this answers is not in the document]
Comment (AshCorr) [reply .github/workflows/dcr-container.yml:40]: It does, but I believe the merge commit behavior only happens if an action is triggered by a `pull_request` event

File: dotcom-rendering/Containerfile (hunk 1 above)
Comment (AshCorr) [reply dotcom-rendering/Containerfile:2]: Added!

**Review by ioannakok — APPROVED** [review]
Looks great! 👏 🌟 🎉

**Comment by ioannakok** [comment]
Lighthouse job has been deactivated until [this](https://github.com/GoogleChrome/lighthouse-ci/issues/886) gets addressed.

**Pull request merged**